common/src/main/java/com/sequenceiq/cloudbreak/quartz/statuschecker/service/StatusCheckerJobService.java

```diff
@@ -76,15 +79,27 @@ public void schedule(Long id, Class<? extends JobResourceAdapter<?>> resource) {
 
     public <T> void scheduleLongIntervalCheck(JobResourceAdapter<T> resource) {
         JobDetail jobDetail = buildJobDetail(resource, Map.of(SYNC_JOB_TYPE, LONG_SYNC_JOB_TYPE));
-        Trigger trigger = buildJobTrigger(jobDetail, statusCheckerConfig.getIntervalInSeconds(), statusCheckerConfig.getLongIntervalInSeconds());
-        schedule(jobDetail, trigger, resource.getLocalId());
+        Trigger trigger = buildJobTrigger(jobDetail, resource.getJobResource(), statusCheckerConfig.getIntervalInSeconds(),
+                statusCheckerConfig.getLongIntervalInSeconds());
+        schedule(jobDetail, trigger, resource.getJobResource());
+    }
+
+    public <T> void scheduleLongIntervalCheck(Long id, Class<? extends JobResourceAdapter<?>> resourceAdapterClass) {
+        try {
+            Constructor<? extends JobResourceAdapter> c = resourceAdapterClass.getConstructor(Long.class, ApplicationContext.class);
+            JobResourceAdapter resourceAdapter = c.newInstance(id, applicationContext);
+            scheduleLongIntervalCheck(resourceAdapter);
+        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException e) {
+            LOGGER.error(String.format("Error during scheduling quartz job: %s", id), e);
```

Comment:
nit: unnecessary String.format, you could simply use `LOGGER.error("Error during scheduling quartz job: {}", id, e)`

Reply:
done